lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -247,7 +247,7 @@ function dosomething_global_convert_country_to_language($country) {
 function dosomething_global_get_language($account, stdClass $node = NULL) {
   if ($node) {
     // If there is a published translation for the node in the user's language, use it.
-    if ($node->translations->data['$account->language']['status']) {
+    if ($node->translations->data{$account->language}['status']) {
```

Comment:
I found this on manual, curly and square braces are interchangeable when accessing array element:
http://php.net/manual/en/language.types.array.php#language.types.array.syntax.accessing
I still won't recommend using this notation as i ist very uncommon.

> It's not an array, it's an object

But you're accessing it here as an array. Did you mean 

```php
if ($node->translations->data->{$account->language}['status']) {
```

Note the `->` after data.